Fix TypeScript exports for classes

lutovich opened the pull request:
They were previously exported in a way that did not allow user code to create instances using `new`. For example:

```typescript
const date = new neo4j.types.Date(1, 2, 3);
```

did not compile.

Added tests for all types exported from `neo4j.types` object.

Fixes #426

lutovich committed:
Fix TypeScript exports for classes

They were previously exported in a way that did not allow user code
to create instances using `new`. For example:

```typescript
const date = new neo4j.types.Date(1, 2, 3);
```

did not compile.

Added tests for all types exported from `neo4j.types` object.

ali-ince reviewed (commented):
I have a small comment around exporting `Node`, `Relationship`, `UnboundRelationship`, `PathSegment`, `Path` and `Record` with constructors as this change will make them part of the public API. Would it be better to keep them as they were?

lutovich committed:
Do not expose all constructors in TS declarations

Node, Relationship, UnboundRelationship, PathSegment, Path and
Record should not be exposed in `neo4j.types` constant.
